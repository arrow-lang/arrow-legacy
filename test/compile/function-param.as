extern def exit(s: int);

def do(a: int) -> int {
  return a;
}

exit(do(42));
