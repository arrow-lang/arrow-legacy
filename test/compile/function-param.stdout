define internal void @function-param..init() {
  %1 = call i128 @do(i128 42)
  call void @exit(i128 %1)
  ret void
}

define internal i128 @do(i128) {
  %2 = alloca i128
  store i128 %0, i128* %2
  %3 = load i128* %2
  ret i128 %3
}

declare void @exit(i128 signext)

define i32 @main(i32, i8**, i8**) {
  call void @function-param..init()
  ret i32 0
}
